toolkit/scripts/srpm_expand.mk

```diff
@@ -17,26 +21,27 @@ clean: clean-expand-specs
 clean-expand-specs:
 	rm -rf $(BUILD_SPECS_DIR)
 	rm -rf $(STATUS_FLAGS_DIR)/build_specs.flag
+	rm -rf $(srpm_expand_logs_dir)
 
 # The directory freshness is tracked with a status flag. The status flag is only updated when all SPECS have been
 # updated.
 $(BUILD_SPECS_DIR): $(STATUS_FLAGS_DIR)/build_specs.flag
 	@touch $@
-	@echo Finished updating $@
+	@echo "Finished updating $@." | tee -a $(srpm_expand_log)
 
+# For each SRPM, if it is newer than the spec extract a new copy of the .spec file
 $(STATUS_FLAGS_DIR)/build_specs.flag: $(srpms) $(BUILD_SRPMS_DIR)
-	# For each SRPM, if it is newer than the spec extract a new copy of the .spec file
+	@echo "Extracting new or updated SRPMs from \"$(BUILD_SRPMS_DIR)\"." | tee $(srpm_expand_log) && \
 	for srpm in $(srpms); do \
-		srpm_file=$${srpm} && \
-		spec_destination=$(BUILD_SPECS_DIR)/$$(rpm -qp $${srpm_file} --define='with_check 1' --queryformat %{NAME}-%{VERSION}-%{RELEASE}/%{NAME}.spec) && \
-		spec_dir=$$(dirname $${spec_destination}) && \
-		if [ $${srpm_file} -nt $@ -o ! -f $@ ]; then \
-			mkdir -p $${spec_dir} && \
-			echo extracting $${spec_destination} && \
-			cd $${spec_dir} && rpm2cpio $${srpm_file} | cpio -idvu || $(call print_error,Failed to expand $${srpm_file}) ; \
-		fi || $(call print_error,If in $@ failed) ; \
-	done || $(call print_error,Loop in $@ failed) ; \
+		spec_destination=$(BUILD_SPECS_DIR)/$$(rpm -qp $$srpm --define='with_check 1' --queryformat %{NAME}-%{VERSION}-%{RELEASE}/%{NAME}.spec) && \
+		spec_dir=$$(dirname $$spec_destination) && \
+		if [ ! -f $@ -o $$srpm -nt $@ ]; then \
+			srpm_filename=$$(basename $$srpm) && \
+			echo "Extracting \"$$srpm_filename\" to \"$$spec_dir\"." | tee -a $(srpm_expand_log) && \
+			mkdir -p $$spec_dir && \
+			cd $$spec_dir && \
+			rpm2cpio $$srpm | cpio -idvu 2>>$(srpm_expand_log) && \
```

Comment:
Is this srpm expansion multi-threaded or single-threaded? If multi-threaded, could multiple `rpm2cpio` happen concurrently, thus interleaving messages in the `srpm_expand_log`?

Reply:
Good question. As much as it pains me to say it, it's single threaded. Making it run in parallel is something I decided to attempt in a different PR. At that point I'll also make sure we don't log into the same file, to keep the logs readable.